### ISSUE TITLE
Automatic new type of layer creation mechanism for PyTorch

### ISSUE THREAD

**Issue by andreiliphd** [pytorch/pytorch#14345]
## 🚀 Feature

 I want to offer you a method that put production of a new type of layers on an industrial scale.

## Motivation

Different industries like Manufacturing, Chemical, Retail and etc in a need for a new type of layers suited for particular industries.

## Pitch

You train your model for a particular task and after applying this feature you have a new type of layer. You just use modularity principle and connect you new layers in the way that brings result.

## Alternatives

A quick search showed that this idea a new one. Please, state here in case you've seen something like that.

## Additional context

HOW: 
1, You trained your network for particular industry application and invested a lot of computational power to make it work. And this neural network work for solving a particular problem in a particular industry very well.
2. You freeze all layers in your neural network.
3. Let's suppose we have an MSE loss for a simplicity.
4. Then you start to training target(label) looking for INTEGRAL of a function of your neural network. It will produce a mathematical formula describing a neural network. Let's call ANTI GRAD algorithm.
5. After it's done you have a mathematical formula that can easily be turned into a new type of layer for particular industry application with a simple(you can tune simplicity) formula for Chemistry, Economics, Medicine.

APPLICATION:
1. Creation of new type of layers suited for particular industries and application.
2. Research work can be enhanced. We are turning a neural network into the classic mathematical language.
